### PR TITLE
Add map integration to website

### PR DESCRIPTION
**Pull request by Philipp000**
As mentioned in issue #27 this includes a map integration into the gh-pages website.

Attached you can see the current version. The map is opened in a modal, once the map button besides the canteens is clicked. Further, it is possible to change the canteen via the map, by clicking on the link in one of the popups.
What do you think about this implementation?

Bildschirmaufnahme 2021-11-18 um 17 52 59

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -70,12 +63,122 @@ function getDate() {
 }
 
 function Controls() {
-    var LocationsDropdown = {
+    let showModal = false;
+    let canteens = [];
+
+    function openStreetMap() {
+        const greenIcon = new L.Icon({
+            iconUrl: 'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-green.png',
```
Comment (COM8) [inline js/app.js:71]: Can't we provide a local copy of this marker?
What is it licensed under?

File: js/app.js [hunk 2]
```diff
@@ -70,12 +63,122 @@ function getDate() {
...
+            shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
```
Comment (COM8) [inline js/app.js:72]: What is this `shadowUrl` for?

File: js/app.js [hunk 3]
```diff
@@ -70,12 +63,122 @@ function getDate() {
...
+            popupAnchor: [1, -34],
+            shadowSize: [41, 41]
+        });
+
+        return {
+            oncreate: function (vnode) {
+                const map = L.map(vnode.dom)
+                    .setView([48.15, 11.55], 10); // coordinates for munich
+                L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+                    attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
```
Comment (COM8) [inline js/app.js:84]: We have to credit OpenStreetMap to use it
> [...] You are free to copy, distribute, transmit and adapt our data, as long as you credit OpenStreetMap and its contributors. [...]

https://www.openstreetmap.org/copyright

**Review by COM8 — CHANGES_REQUESTED** [review]
Nice idea with the popup!
A few things are still missing.
Also could we introduce a button which locates the user and then switches to the nearest canteen? But only if the user clicks on the button to prevent the website from requesting the users location per default.

File: js/app.js (hunk 3 above)
Comment (Philipp000) [reply js/app.js:84]: I used the example code from leaflet, which includes the Copyright mention here in line 84, don't you think, this is sufficient?
It looks like this on the webpage (bottom right corner):
Bildschirmfoto 2021-11-20 um 17 36 41

File: js/app.js (hunk 1 above)
Comment (Philipp000) [reply js/app.js:71]: The GitHub repository is licensed under BSD (2-clause) https://github.com/pointhi/leaflet-color-markers/blob/master/LICENSE

So a local copy is possible. However, I'm not sure, on how to generally include the copyright notice?

File: js/app.js (hunk 3 above)
Comment (COM8) [reply js/app.js:84]: Aha! Overlooked that. that's fine 👍.

File: js/app.js (hunk 1 above)
Comment (Philipp000) [reply js/app.js]: I have now changed it, and use CSS instead of the marker from this external repository

File: js/app.js (hunk 2 above)
Comment (Philipp000) [reply js/app.js]: I think, it was. concercing the shadow of the marker, which makes it look 3D.
However, i used CSS now instead of this icon.

**Comment by Philipp000** [comment]
I have now included a button, to select the closest canteen.
Further i fixed the issues with the green marker.
Additionally, I included the source code for leaflet in the repository, to keep it consistent with the other libraries.

Currently, it looks like this: 
Bildschirmaufnahme 2021-11-22 um 18 24 12

Do you think, there is something else to do?
Currently, I always set the initial view, so that one can see munich. I could also set it, so that the currently selected canteen is in the center. Which do you think is better?

**Comment by COM8** [comment]
I think that should be everything then.
Setting the initial view to munich is fine.

**Review by COM8 — APPROVED** [review]

File: js/app.js (hunk 2 above)
Comment (COM8) [reply js/app.js]: Aha, ok :)

**Review by COM8 — APPROVED** [review]
Well done. Thanks!

**Pull request merged**